src/app/upgrade.h

```diff
@@ -26,279 +26,11 @@
-
-    if (ask && !userAcceptsUpgrade()) return false;
-
-    const QStringList backupFiles = backupFolderDir.entryList(
-                QStringList(QLatin1String("*.fastresume")), QDir::Files, QDir::Unsorted);
-    const QRegularExpression rx(QLatin1String("^([A-Fa-f0-9]{40})\\.fastresume$"));
-    for (const QString &backupFile : backupFiles) {
-        const QRegularExpressionMatch rxMatch = rx.match(backupFile);
-        if (rxMatch.hasMatch()) {
-            const QString hashStr = rxMatch.captured(1);
-            if (upgradeResumeFile(backupFolderDir.absoluteFilePath(backupFile), oldResumeData[hashStr].toHash()))
-                oldResumeData.remove(hashStr);
-            else
-                Logger::instance()->addMessage(QObject::tr("Couldn't migrate torrent with hash: %1").arg(hashStr), Log::WARNING);
-        }
-        else {
-            Logger::instance()->addMessage(QObject::tr("Couldn't migrate torrent. Invalid fastresume file name: %1").arg(backupFile), Log::WARNING);
-        }
-    }
-
-    for (auto i = oldResumeData.cbegin(); i != oldResumeData.cend(); ++i) {
-        const QVariantHash oldTorrent = i.value().toHash();
-        if (oldTorrent.value("is_magnet", false).toBool()) {
-            const QString &hash = i.key();
-            libtorrent::entry resumeData;
-            resumeData["qBt-magnetUri"] = oldTorrent.value("magnet_uri").toString().toStdString();
-            resumeData["qBt-paused"] = false;
-            resumeData["qBt-forced"] = false;
-
-            resumeData["qBt-savePath"] = oldTorrent.value("save_path").toString().toStdString();
-            resumeData["qBt-ratioLimit"] = static_cast<int>(oldTorrent.value("max_ratio", -2).toReal() * 1000);
-            resumeData["qBt-label"] = oldTorrent.value("label").toString().toStdString();
-            resumeData["qBt-name"] = oldTorrent.value("name").toString().toStdString();
-            resumeData["qBt-seedStatus"] = oldTorrent.value("seed").toBool();
-            resumeData["qBt-tempPathDisabled"] = false;
-
-            int queuePosition = oldTorrent.value("priority", 0).toInt();
-            resumeData["qBt-queuePosition"] = (queuePosition >= 0 ? queuePosition : 0);
-
-            QString filename = QString("%1.fastresume").arg(hash);
-            QString filepath = backupFolderDir.absoluteFilePath(filename);
-
-            QFile resumeFile(filepath);
-            QVector<char> out;
-            libtorrent::bencode(std::back_inserter(out), resumeData);
-            if (resumeFile.open(QIODevice::WriteOnly))
-                resumeFile.write(&out[0], out.size());
-        }
-    }
-
-    int counter = 0;
-    QString backupResumeFilename = oldResumeFilename + ".bak";
-    while (QFile::exists(backupResumeFilename)) {
-        ++counter;
-        backupResumeFilename = oldResumeFilename + ".bak" + QString::number(counter);
-    }
-    QFile::rename(oldResumeFilename, backupResumeFilename);
+    // Intentionally left no-op as a placeholder for future use
```

Comment:
we should move this to .cpp file, optional for this PR but IMO required in later use.

Reply:
Why?

Reply:
to save unnecessary compilation work.

Reply:
Both "main.cpp" and "upgrade.h" are rarely changeable, so there will be no significant gain.

Reply:
>Let's see, before this PR upgrade.h also includes 6 local headers which means main.cpp will get recompiled every time when these headers AND its dependencies headers changed.

Not so important in the case of a once-included header. One file (`main.cpp` or `upgrade.cpp`) will be recompiled anyway...
I just want to point out the exclusivity of such cases, when we extract some code not to make it shared, but only to separate some  logically isolated part. In such cases, we may have it in the header.
But if you insist on having it in .cpp, I won't resist.